programs/librapay_api/src/librapay_transaction.rs

```diff
@@ -0,0 +1,39 @@
+use solana_sdk::hash::Hash;
+use solana_sdk::pubkey::Pubkey;
+use solana_sdk::transaction::Transaction;
+use solana_sdk::signature::{Keypair, KeypairUtil};
+use crate::librapay_instruction;
+
+pub fn payment(program_id: &Pubkey, from: &Keypair, to: &Pubkey, lamports: u64, recent_blockhash: Hash) -> Transaction {
+    let ix = librapay_instruction::payment(program_id, &from.pubkey(), to, lamports);
+    Transaction::new_signed_with_payer(vec![ix], Some(&from.pubkey()), &[from], recent_blockhash)
+}
+
+#[cfg(test)]
+mod tests {
+    use solana_sdk::genesis_block::create_genesis_block;
+    use solana_sdk::signature::{Keypair, KeypairUtil};
+    use solana_runtime::bank::Bank;
+    use super::*;
+    use crate::upload_move_program;
+    use solana_runtime::bank_client::BankClient;
+    use std::sync::Arc;
+
+    fn create_bank(lamports: u64) -> (Arc<Bank>, Keypair, Pubkey) {
+        let (genesis_block, mint_keypair) = create_genesis_block(lamports);
+        let mut bank = Bank::new(&genesis_block);
+        bank.add_instruction_processor(solana_move_loader_api::id(), solana_move_loader_api::process_instruction);
+        let shared_bank = Arc::new(bank);
+        let bank_client = Arc::new(BankClient::new_shared(&shared_bank));
+        let program_pubkey = upload_move_program(&mint_keypair, &bank_client);
+        (shared_bank, mint_keypair, program_pubkey)
+    }
+
+    #[test]
+    fn test_payment() {
+        let (bank, alice_keypair, program_id) = create_bank(42);
+        let to = Keypair::new();
+        let tx = payment(&program_id, &alice_keypair, &to.pubkey(), 1, bank.last_blockhash());
+        bank.process_transaction(&tx).unwrap();
```

Comment:
@jackcmay, check this out. This is a really good test case. We can easily query the bank for `to.pubkey()` just after it and see if that Account's data represents a successful payment. Can you advise on how to write that code? I assume the e2e helper functions?

Reply:
Looks like this test: https://github.com/solana-labs/solana/blob/8a12ed029cfa38d4a45400916c2463fb82bbec8c/programs/move_loader_api/src/lib.rs#L299